org.scala-ide.sdt.core/src/org/scalaide/ui/editor/DecoratedInteractiveEditor.scala

```diff
@@ -19,10 +21,21 @@ trait DecoratedInteractiveEditor extends ISourceViewerEditor {
   private var previousAnnotations = List[Annotation]()
 
   /**
+   * This removes all annotations in the region between `start` and `end`.
+   */
+  def removeAnnotationsInRegion(start: Int, end: Int): Unit = annotationModel foreach { model ⇒
+    val annsToRemove = model.getAnnotationIterator.asScala.filter { ann ⇒
```

Comment:
You could use `IAnnotationModelExtension2.getAnnotationIterator(offset,length,...)`. It might be more efficient than linear search.